### PR TITLE
OAuth2.0 Password Grant flow tests

### LINKED ISSUE #6: Unsupported grant type returns error
Described at https://github.com/stormpath/stormpath-framework-spec/blob/master/oauth2.md#weboauth2enabled

> If enabled and the grant type is not `client_credentials` or `password`, then we should return the OAuth-compliant error code `unsupported_grant_type`

### LINKED ISSUE #7: Missing grant type returns error
Described in https://github.com/stormpath/stormpath-framework-spec/blob/master/oauth2.md#weboauth2enabled

> If no grant type is specified, the error should be `invalid_request`

### LINKED ISSUE #11: Password grant flow with username/password
Described at https://github.com/stormpath/stormpath-framework-spec/blob/master/oauth2.md#password-grant-flow

### LINKED ISSUE #15: POST must include form parameters
The OAuth2 route should return an error for POST requests that do not include form parameters, or contain some other type of body.

### LINKED ISSUE #16: GET should return 405
The OAuth2 route should return `HTTP 405` for a GET request.

### LINKED ISSUE #18: Password grant flow with email/password
Described at https://github.com/stormpath/stormpath-framework-spec/blob/master/oauth2.md#password-grant-flow

> Although the Oauth2 spec requires the parameter to be named `username`, this value can also be the email address of a Stormpath account.

### LINKED ISSUE #27: Create account via JSON
Described in https://github.com/stormpath/stormpath-framework-spec/blob/master/registration.md#-post-body-format

```
POST /register
Content-Type: application/json
{
    "email": "test@stormpath.com",
    "password": "changeme",
    "givenName": "Test",
    "surname": "Testerman"
}
```

should return

```
HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8
{
  "account": {
    {
      "href": "https://api.stormpath.com/v1/accounts/xxx",
      "username": "test@stormpath.com",
      "modifiedAt": "2016-01-26T20:50:03.931Z",
      "status": "ENABLED",
      "createdAt": "2015-10-13T20:54:22.215Z",
      "email": "test@stormpath.com",
      "middleName": null,
      "surname": "Testerman",
      "givenName": "Test"
      "fullName": "Test Testerman"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by nbarbettini**
I wrote some basic tests to exercise the `/oauth/tokens` endpoint.

I'm just getting the hang of writing tests in groovy and RestAssured, so any feedback from @dogeared @mrioan @lhazlewood is appreciated.

Fixes #6 
Fixes #7 
Fixes #16 
Fixes #15 
Fixes #27 
Fixes #11 
Fixes #18

### CODE REVIEW

**Comment by nbarbettini** [comment]
Btw, these are running live against ASP.NET :smile:

**Comment by lhazlewood** [comment]
Awesome!

**Comment by nbarbettini** [comment]
No longer in progress; ready for review!

**Comment by edjiang** [comment]
I should be ignoring `LoginJSONIT` correct?

**Comment by nbarbettini** [comment]
Yes, git fail on my part (sorry).

File: src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+                .body(jsonAsMap)
+            .when()
+                .post(registerRoute)
+            .then()
+                .statusCode(200)
+            .extract()
+                .path("account.href")
+
+        this.accountHref = createdHref
+        deleteOnClassTeardown(createdHref)
```
Comment (edjiang) [inline src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy:64]: Not too familiar with testing idioms here, but if we're going to use `createTestAccount` everywhere, maybe it should be in a utility?

**Comment by nbarbettini** [comment]
Btw: don't merge this until I have a chance to rebase it.

File: src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy (hunk 1 above)
Comment (nbarbettini) [inline src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy:64]: Probably, not entirely sure yet how we want to handle this. Should be a `todo` item or an issue for later work in the TCK.

Comment (nbarbettini) [inline src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy:64]: Same goes for the `private static final String registerPath = "/register"` stuff -- probably should be centralized at one point so we can be DRY :smile:

**Comment by edjiang** [comment]
OK, looks good to me. Feel free to rebase and merge in, @nbarbettini. Please keep in mind that mobile is not fully supported until we can fix #205

**Comment by nbarbettini** [comment]
Thanks @edjiang and thanks for creating the issue for Refresh grant. :+1: It slipped my mind before.

File: src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy (hunk 1 above)
Comment (nbarbettini) [inline src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy:64]: Created #209, thanks @edjiang

**Pull request merged**